core/ffmpeg/hlsVariantWriteMonitor.go

```diff
@@ -0,0 +1,95 @@
+package ffmpeg
+
+import (
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/net/context"
+)
+
+type hlsVariantWriteMonitor struct {
+	playlistEvents chan hlsWriteEvent
+	segmentEvents  chan hlsWriteEvent
+	logger         *log.Logger
+}
+
+type hlsWriteEvent struct {
+	Dir  string
+	Time time.Time
+}
+
+func newHlsVariantWriteMonitor(ctx context.Context, logger *log.Logger, notificationThreshold time.Duration) *hlsVariantWriteMonitor {
```

Comment:
I had to accept a logger here so I could write tests.  It's a bit out of the norm since it doesn't use the logrus default methods.

Additionally I added a context in case this needs clean shutdown/reload later on.